pkg/datapath/linux/node.go

```diff
@@ -689,6 +689,9 @@ func getSrcAndNextHopIPv4(nodeIPv4 net.IP) (srcIPv4, nextHopIPv4 net.IP, err err
 // this case it does not bail out early if the ARP entry already exists, and
 // sends the ARP request anyway.
 func (n *linuxNodeHandler) insertNeighbor(ctx context.Context, newNode *nodeTypes.Node, refresh bool) {
+	if !option.Config.EnableL2NeighDiscovery {
```

Comment:
I think https://github.com/cilium/cilium/blob/master/pkg/datapath/linux/node.go#L1378 is a better place to opt out from the L2 neigh discovery.

Reply:
It seems some of the functions on this interface https://github.com/cilium/cilium/blob/a8b34806e2905c675b90b61f5c9d8ae7609057f2/pkg/datapath/node.go#L118-L147 do L2 neighbor discovery, specifically `NodeAdd`, `NodeUpdate` and `NodeConfigurationChanged`(whose implementation you are recommending I modify) and the [manager](https://github.com/cilium/cilium/blob/a8b34806e2905c675b90b61f5c9d8ae7609057f2/pkg/node/manager/manager.go) seems to call `NodeAdd` and `NodeUpdate` methods, so making the change in `NodeConfigurationChanged` might not adequate... Or do I misunderstand the code in the various places I have linked?

Reply:
The methods you mentioned do the L2 related stuff only if `enableNeighDiscovery == true`. My linked code sets the latter. So you could just add another case `!option.Config.EnableL2NeighDiscovery` which would set `enableNeighDiscovery = false`.

Reply:
Thanks for the feedback, that makes sense to me.

Reply:
Addressed in https://github.com/cilium/cilium/commit/9da5b893a1226af2bb57a730fd3003cbfdffb0b7